standalone-metastore/metastore-server/src/main/java/org/apache/hadoop/hive/metastore/HMSHandler.java

```diff
@@ -8994,10 +9028,15 @@ public boolean set_aggr_stats_for(SetPartitionsStatsRequest request) throws TExc
             colNames, newStatsMap, request);
       } else { // No merge.
         Table t = getTable(catName, dbName, tableName);
-        for (Map.Entry<String, ColumnStatistics> entry : newStatsMap.entrySet()) {
-          // We don't short-circuit on errors here anymore. That can leave acid stats invalid.
-          ret = updatePartitonColStatsInternal(t, entry.getValue(),
-              request.getValidWriteIdList(), request.getWriteId()) && ret;
+        // We don't short-circuit on errors here anymore. That can leave acid stats invalid.
+        if (newStatsMap.size() > 1) {
+          LOG.info("ETL_PERF started updatePartitionColStatsInBatch");
+          ret = updatePartitionColStatsInBatch(t, newStatsMap,
+                  request.getValidWriteIdList(), request.getWriteId());
+          LOG.info("ETL_PERF done updatePartitionColStatsInBatch");
+        } else {
+          ret = updatePartitonColStatsInternal(t, newStatsMap.values().iterator().next(),
```

Comment:
The batching is a specific condition, when the partitions are from same table. I am not sure if in other flows this holds good. So i have used the batching only in the flow where this is guaranteed.

Reply:
Won't updatePartitionColStatsInBatch work with a batch of size 1? This complicates code and would cause some confusion. Also looks like updatePartitonColStatsInternal requires new DB txn unlike the  updatePartitionColStatsInBatch.

Reply:
Yes .. updatePartitonColStatsInternal requires a txn as it uses JDO.

Reply:
in future we would need to support both implementations, why can't we generalize?